controllers/front/ListComments.php

```diff
@@ -64,20 +64,20 @@ public function display()
             $productComment['content'] = htmlentities($productComment['content']);
             $productComment['date_add'] = $dateFormatter->format($dateAdd);
 
-            if ($isLastNameAnonymous && isset($productComment['lastname'])) {
-                $productComment['lastname'] = substr($productComment['lastname'], 0, 1) . '.';
-            }
+            // The customer has firstname and lastname, for guest we only have customer_name field
+            $productComment['customer_name'] = !empty($productComment['customer_name'])
+                ? $productComment['customer_name']
+                : $productComment['firstname'] . ' ' . $productComment['lastname'];
 
-            // if registered customer : return customer first and last name instead of using customer_name
-            if (!empty($productComment['lastname'])) {
-                $productComment['customer_name'] = htmlentities($productComment['firstname'] . ' ' . $productComment['lastname']);
-            } else {
-                $productComment['customer_name'] = htmlentities($productComment['customer_name']);
+            if ($isLastNameAnonymous) {
+                $productComment['customer_name'] = $this->anonymizeName($productComment['customer_name']);
             }
 
+            $product['customer_name'] = htmlentities($productComment['customer_name']);
```

Comment:
I think you used wrong variable here.
```suggestion
            $productComment['customer_name'] = htmlentities($productComment['customer_name']);
```

Reply:
@micka-fdz nice catch!